android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationHelper.java

```diff
@@ -406,6 +411,35 @@ public void clearNotifications() {
         notificationManager.cancelAll();
     }
 
+    public void clearDeliveredNotifications(ReadableArray identifiers) {
+      NotificationManager notificationManager = notificationManager();
+      for (int index = 0; index < identifiers.size(); index++) {
+        String id = identifiers.getString(index);
+        Log.i(LOG_TAG, "Removing notification with id " + id);
+        notificationManager.cancel(Integer.parseInt(id));
+      }
+    }
+
+    public WritableArray getDeliveredNotifications() {
+      NotificationManager notificationManager = notificationManager();
+      StatusBarNotification delivered[] = notificationManager.getActiveNotifications();
+      Log.i(LOG_TAG, "Found " + delivered.length + " delivered notifications");
+      WritableArray result = Arguments.createArray();
+      for (StatusBarNotification notification : delivered) {
+        Notification original = notification.getNotification();
+        Bundle extras = original.extras;
+        WritableMap notif = Arguments.createMap();
+        notif.putString("identifier", "" + notification.getId());
+        notif.putString("title", extras.getString(Notification.EXTRA_TITLE));
+        notif.putString("body", extras.getString(Notification.EXTRA_TEXT));
+        notif.putString("tag", notification.getTag());
+        notif.putString("group", original.getGroup());
+        result.pushMap(notif);
```

Comment:
I tried to stay consistent to the return structure in https://facebook.github.io/react-native/docs/pushnotificationios.html#getdeliverednotifications

but there is no such thing as a 'userInfo' and I couldn't figure out how to add custom metadata to the notification when building it originally (https://developer.android.com/reference/android/app/Notification.Builder.html)

Reply:
I'd site that link in the code itself - it'll point out later on why we decided to return this structure.